kotlin/vector-tiled-layer-custom-style/src/main/java/com/esri/arcgisruntime/sample/arcgisvectortiledlayercustomstyle/MainActivity.kt

```diff
@@ -0,0 +1,315 @@
+            100000000.0
+        )
+        setMap(vectorTiledLayer, viewpoint)
+    }
+
+    /**
+     *  Checks local cache for local cache files using the [itemID].
+     *  If not it calls loadLayerWithOfflineCustomStyle([itemID]) to retrieve the cache files.
+     */
+    private fun checkOfflineItemCache(itemID: String) {
+
+        val portalItem = PortalItem(Portal("https://www.arcgis.com"), itemID)
+        val itemResourceCache =
+            ItemResourceCache(getExternalFilesDir(null)?.path + "/" + portalItem.itemId)
+        itemResourceCache.addDoneLoadingListener {
+            if (itemResourceCache.loadStatus == LoadStatus.LOADED) {
+                setResourceAndVectorTileCache(itemResourceCache)
+            } else {
+                loadLayerWithOfflineCustomStyle(itemID)
+            }
+        }
+        itemResourceCache.loadAsync()
+    }
+
+    /**
+     * Retrieves the style resource files using [itemID] and caches it to the local device.
+     */
+    private fun loadLayerWithOfflineCustomStyle(itemID: String) {
+
+        // Retrieve the layer from online.
+        val portalItem = PortalItem(Portal("https://www.arcgis.com"), itemID)
+        val task = ExportVectorTilesTask(portalItem)
+
+        val exportVectorTilesJob =
+            task.exportStyleResourceCache(getExternalFilesDir(null)?.path + "/" + portalItem.itemId)
+        exportVectorTilesJob.addJobDoneListener {
+            if (exportVectorTilesJob.status == Job.Status.SUCCEEDED) {
+                setResourceAndVectorTileCache(exportVectorTilesJob.result.itemResourceCache)
+            } else {
+                Toast.makeText(
+                    this,
+                    "Error reading cache: " + exportVectorTilesJob.error.message,
+                    Toast.LENGTH_LONG
+                ).show()
+            }
+        }
+        exportVectorTilesJob.start()
+
+    }
+
+    /**
+     * Sets up an ArcGISVectorTiledLayer using the [itemResourceCache]
+     * for the setMap() function.
+     */
+    private fun setResourceAndVectorTileCache(itemResourceCache: ItemResourceCache) {
+        //Loads the vector tile layer cache.
+        val vectorTileCache = VectorTileCache(getExternalFilesDir(null)?.path + "/dodge_city.vtpk")
+        vectorTileCache.loadAsync()
+        vectorTileCache.addDoneLoadingListener {
+            if (vectorTileCache.loadError != null)
+                Log.e("VectorTileCache: ", vectorTileCache.loadError.message.toString())
+            else {
+                // Loads the layer based on the vector tile cache and the style resource.
+                val layer = ArcGISVectorTiledLayer(vectorTileCache, itemResourceCache)
+                layer.addDoneLoadingListener {
+                    if (layer.loadError != null)
+                        Log.e("VectorTiledLayer: ", layer.loadError.toString())
+                    else
+                        Log.d("VectorTiledLayer:", "Loaded successfully")
```

Comment:
Let's pass the error as a toast as well and not bother with the loaded successfully debug line